Detectors/PHOS/calib/include/PHOSCalib/CalibDB.h

```diff
@@ -0,0 +1,219 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#include <exception>
+#include <map>
+#include <string>
+#include "Rtypes.h"
+#include "RStringView.h"
+#include "CCDB/CcdbApi.h"
+
+namespace o2
+{
+
+namespace phos
+{
+
+class BadChannelMap;
+class CalibParams;
+
+/// \class CalibDB
+/// \brief Interface to calibration data from CCDB for PHOS
+/// \author Dmitri Peresunko, adopted from EMCAL (Markus Fasel)
+/// \since Aug 1, 2019
+///
+/// Interface handling simple access to CCDB content for
+/// PHOS objects. The interface allows storing and
+/// reading of the common PHOS calibration objects
+/// - Bad Channel Map
+/// - Gain, HG/LG ratio, Time calibration
+/// - Run-by-run gaincorrections
+/// Users only need to specify the CCDB server, the timestamp and
+/// (optionally) additional meta data. Handling of the CCDB path
+/// and type conversions is done internally - users deal directly
+/// with the low level containers.
+///
+/// Attention: The read process of the CCDB objects might fail,
+/// either because the query of the CCDB was not successfull
+/// (wrong server / path / timestamp) or the object type is different
+/// and internal conversion failed. In both cases dedicated exceptions
+/// are thrown:
+/// - ObjectNotFoundException in case of failure of the query
+/// - TypeMismatchException in case object type doesn't match the expected type
+/// Users must handle the exceptions.
+class CalibDB
```

Comment:
This should and will be offered on a higher level (currently working on it). I am ok to merge now but we should target to reduce this code duplication later on.